### PR TITLE
HDDS-3093. Allow forced overwrite of local file

### PR DESCRIPTION
**Pull request by adoroszlai**
## What changes were proposed in this pull request?

Add `-f` (`--force`) flag for `ozone sh key get`, to allow force overwrite of local file.  This is useful for testing:

 * download the same key multiple times without having to manually delete the target
 * download to `/dev/null` (which already exists), only to verify existence and checksum of the key

Hash cannot be calculated for `/dev/null`.

This PR is based on top of #775, so that one should be merged first.  The only change here is 2ca1fe23cf72eb813154c17260868b6ecd030801.

https://issues.apache.org/jira/browse/HDDS-3093

## How was this patch tested?

```
$ ozone freon ockg -n 1 -p test
...

$ ozone sh key get /vol1/bucket1/test/0 /tmp/asdf

$ ozone sh key get /vol1/bucket1/test/0 /tmp/asdf
/tmp/asdf exists. Download would overwrite an existing file. Aborting.

$ ozone sh key get --force /vol1/bucket1/test/0 /tmp/asdf

$ ozone sh --verbose key get --force /vol1/bucket1/test/0 /tmp/asdf
Volume Name : vol1
Bucket Name : bucket1
Key Name : test/0
Downloaded file hash : 528e4f96633a453d18ee01fdb5d7c536

$ ozone sh --verbose key get --force /vol1/bucket1/test/0 /dev/null
Volume Name : vol1
Bucket Name : bucket1
Key Name : test/0
```

Added smoke test for overwrite case.

### CODE REVIEW

File: hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/shell/keys/GetKeyHandler.java [hunk 1]
```diff
@@ -80,7 +88,7 @@ protected void execute(OzoneClient client, OzoneAddress address)
       IOUtils.copyBytes(input, output, chunkSize);
     }
 
-    if (isVerbose()) {
+    if (isVerbose() && !"/dev/null".equals(dataFile.getAbsolutePath())) {
```
Comment (xiaoyuyao) [inline hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/shell/keys/GetKeyHandler.java:91]: Hardcoded path like "/dev/null" may not work in other platforms like Windows.

Comment (adoroszlai) [reply hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/shell/keys/GetKeyHandler.java:91]: I think that's OK.    This check does not affect functionality and we're not relying on any specific property of `/dev/null` here.  The worst thing that can happen is that if someone issues `ozone sh --verbose key get ... /dev/null` on Windows, hash is not printed.  But I doubt that's a valid file on Windows anyway.

**Comment by adoroszlai** [comment]
Thanks @xiaoyuyao for the review.

File: hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/shell/keys/GetKeyHandler.java (hunk 1 above)
Comment (xiaoyuyao) [reply hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/shell/keys/GetKeyHandler.java:91]: Sounds good to me. +1

**Comment by adoroszlai** [comment]
Thanks @dineshchitlangia for reviewing and merging it.

**Pull request merged**